x/auction/keeper/msg_server_start_auction.go

```diff
@@ -0,0 +1,35 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cast"
+
+	"github.com/Stride-Labs/stride/v5/x/auction/types"
+)
+
+// Get the name of the target pool, if the end-block is already before the current blockchain height, then we can start a new auction
+
+func (k msgServer) StartAuction(goCtx context.Context, msg *types.MsgStartAuction) (*types.MsgStartAuctionResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	now := cast.ToUint64(ctx.BlockHeight())
+	ctx.Logger().Info(fmt.Sprintf("[auction] Request to start auction for pool %d at block %d", msg.GetPoolID(), now))
+
+	auctionPools := k.Keeper.GetAllAuctionPool(ctx)
+	for _, pool := range auctionPools {
```

Comment:
We should remove this for loop and use GetAuctionPool function instead to get auction pool by id.